### PR TITLE
Fixed Group By bug on address data chart

### PR DESCRIPTION
**Pull request by mkingori**
Solves #1734

Added a grouping logic for the graph as per the oldest address transaction block time as follows:
- `< 10 days` ago to group by `All`
- `< 70 days` ago to group by `Day`
- `< 12 months` ago to group by `Week`
- `< 5 years` ago to group by `Month`
- `> 5 years` ago to group by `Year`

### CODE REVIEW

File: db/cache/addresscache.go [hunk 1]
```diff
@@ -571,6 +571,31 @@ func (d *AddressCacheItem) SetBalance(block BlockID, balance *dbtypes.AddressBal
 	d.balance = balance
 }
 
+// OldestAddressTransaction returns data of the oldest address transaction
+func (d *AddressCacheItem) OldestAddressTransaction() (*dbtypes.AddressRowCompact, error) {
+
+	var oldestTxBlockTime int64
+	oldestTransaction := &dbtypes.AddressRowCompact{}
+
+	if d.rows == nil {
```
Comment (chappjc) [inline db/cache/addresscache.go:580]: Not thread-safe

File: db/dcrpg/pgblockchain.go [hunk 2]
```diff
@@ -6324,3 +6324,8 @@ func (pgb *ChainDB) MixedUtxosByHeight() (heights, utxoCountReg, utxoValueReg, u
 	return
 
 }
+
+// OldestTransaction queries the database for the data of the oldest address transaction
```
Comment (chappjc) [inline db/dcrpg/pgblockchain.go:6328]: Does not query the database though. Is that what you intend?

File: views/address.tmpl [hunk 3]
```diff
@@ -152,11 +152,11 @@
...
-                  <button name="year">Year</button>
-                  <button class="btn-selected" name="month">Month</button>
-                  <button name="week">Week</button>
-                  <button name="day" data-fixed="1">Day</button>
-                  <button name="all" data-fixed="1">Block</button>
+                  <button {{if eq $.GroupBy "year"}} class="btn-selected" {{end}} name="year">Year</button>
+                  <button {{if eq $.GroupBy "month"}} class="btn-selected" {{end}} name="month">Month</button>
+                  <button {{if eq $.GroupBy "week"}} class="btn-selected" {{end}} name="week">Week</button>
+                  <button {{if eq $.GroupBy "day"}} class="btn-selected" {{end}} name="day" data-fixed="1">Day</button>
+				          <button{{if eq $.GroupBy "all"}} class="btn-selected" {{end}} name="all" data-fixed="1">Block</button>
```
Comment (chappjc) [inline views/address.tmpl:159]: Mixing tabs and spaces.

File: db/dcrpg/pgblockchain.go (hunk 2 above)
Comment (mkingori) [reply db/dcrpg/pgblockchain.go:6328]: I am just getting the oldest transaction of an address from the cache.

**Comment by mkingori** [comment]
I've made the changes.

**Comment by dnldd** [comment]
I'm grouping by block in the image below but nothing on the chart shows I am, shouldn't the x-axis be the block height instead of the time there?

group-by-block

**Comment by mkingori** [comment]
I believe its correct as it is. The objective was to show when the transactions of a particular address occurred. The block height won't be much relevant here.

**Comment by dnldd** [comment]
Labelling that `group by` option as `block` seems misleading to me. Personally I'd expect to see the block heights if we are showing when the transactions of a particular address occurred relative to blocks. Perhaps the label should be `exact` which would denote the exact time the transaction for the address occurred. That'd fit well with the current graph I think.

**Comment by raedah** [comment]
The way we have dealt with this on other /charts is by having a separate widget to toggle the x axis display. That would probably work here. Also on the other charts we call these BINS instead of GROUP BY, so perhaps better to standardize on that.

**Review by dnldd — APPROVED** [review]

**Pull request closed without merging**